### PR TITLE
Enhancements for the HCP Terraform Operator for Kubernetes

### PR DESCRIPTION
**Pull request by BrianMMcClain**
### What

Document updates to the AgentPool scaling options in the Kubernetes Operator

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. HCP Terraform ).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 1]
```diff
@@ -50,8 +51,24 @@ _Appears in:_
 | --- | --- |
 | `maxReplicas` _integer_ | MaxReplicas is the maximum number of replicas for the Agent deployment. |
 | `minReplicas` _integer_ | MinReplicas is the minimum number of replicas for the Agent deployment. |
-| `targetWorkspaces` _[TargetWorkspace](#targetworkspace)_ | TargetWorkspaces is a list of HCP Terraform workspaces which the agent pool should scale up to meet demand. When this field is ommited the autoscaler will target all workspaces that are associated with the AgentPool. |
+| `targetWorkspaces` _[TargetWorkspace](#targetworkspace)_ | TargetWorkspaces is a list of HCP Terraform Workspaces which<br />the agent pool should scale up to meet demand. When this field<br />is ommited the autoscaler will target all workspaces that are<br />associated with the AgentPool. |
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:54]: why are we manually adding line breaks here?

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 2]
```diff
@@ -50,8 +51,24 @@ _Appears in:_
...
 | `cooldownPeriodSeconds` _integer_ | CooldownPeriodSeconds is the time to wait between scaling events. Defaults to 300. |
+| `cooldownPeriod` _[AgentDeploymentAutoscalingCooldownPeriod](#agentdeploymentautoscalingcooldownperiod)_ | CoolDownPeriod configures the period to wait between scaling up and scaling down |
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:56]:
```suggestion
| `cooldownPeriod` _[AgentDeploymentAutoscalingCooldownPeriod](#agentdeploymentautoscalingcooldownperiod)_ | CoolDownPeriod is the period to wait between scaling up and scaling down |
```

so it's consistent with the other attributes

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 3]
```diff
@@ -50,8 +51,24 @@ _Appears in:_
...
+
+
+#### AgentDeploymentAutoscalingCooldownPeriod
+
+
+
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:62]:
```suggestion

```

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 4]
```diff
@@ -41,7 +41,8 @@ _Appears in:_
 
 
 
-AgentDeploymentAutoscaling allows you to configure the operator to scale the deployment for an AgentPool up and down to meet demand.
+AgentDeploymentAutoscaling allows you to configure the operator
+to scale the deployment for an AgentPool up and down to meet demand.
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:45]:
```suggestion
AgentDeploymentAutoscaling configures the operator to scale the deployment for an AgentPool up and down to meet demand.
```

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 5]
```diff
@@ -50,8 +51,24 @@ _Appears in:_
...
+| `targetWorkspaces` _[TargetWorkspace](#targetworkspace)_ | TargetWorkspaces is a list of HCP Terraform Workspaces which<br />the agent pool should scale up to meet demand. When this field<br />is ommited the autoscaler will target all workspaces that are<br />associated with the AgentPool. |
 | `cooldownPeriodSeconds` _integer_ | CooldownPeriodSeconds is the time to wait between scaling events. Defaults to 300. |
+| `cooldownPeriod` _[AgentDeploymentAutoscalingCooldownPeriod](#agentdeploymentautoscalingcooldownperiod)_ | CoolDownPeriod configures the period to wait between scaling up and scaling down |
+
+
+#### AgentDeploymentAutoscalingCooldownPeriod
+
+
+
+AgentDeploymentAutoscalingCooldownPeriod configures the period to wait between scaling up and scaling down
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:63]:
```suggestion
AgentDeploymentAutoscalingCooldownPeriod configures the period to wait between scaling up and scaling down.
```

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 6]
```diff
@@ -81,8 +98,8 @@ AgentPool is the Schema for the agentpools API.
 | --- | --- |
 | `apiVersion` _string_ | `app.terraform.io/v1alpha2`
 | `kind` _string_ | `AgentPool`
-| `kind` _string_ | Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. [More information](https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds) |
-| `apiVersion` _string_ | APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. [More information](https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources) |
+| `kind` _string_ | Kind is a string value representing the REST resource this object represents.<br />Servers may infer this from the endpoint the client submits requests to.<br />Cannot be updated.<br />In CamelCase.<br />More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds |
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:101]: same thing here, why are we adding line breaks to this?

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 7]
```diff
@@ -91,15 +108,15 @@ AgentPool is the Schema for the agentpools API.
...
 
 _Appears in:_
 - [AgentPool](#agentpool)
 
 | Field | Description |
 | --- | --- |
-| `name` _string_ | Agent Pool name. [More information](/terraform/cloud-docs/agents/agent-pools) |
-| `organization` _string_ | Organization name where the Workspace will be created. [More information](/terraform/cloud-docs/users-teams-organizations/organizations) |
+| `name` _string_ | Agent Pool name.<br />More information:<br />  - https://developer.hashicorp.com/terraform/cloud-docs/agents/agent-pools |
+| `organization` _string_ | Organization name where the Workspace will be created.<br />More information:<br />  - https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/organizations |
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:119]: i think the previous one was ok?

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 8]
```diff
@@ -110,7 +127,12 @@ _Appears in:_
 
 #### AgentToken
 
-HCP Terraform agents use secret tokens, called `AgentToken`s, to connect to the HCP Terraform agent pool. In `spec` only the field `Name` is allowed, the rest are used in `status`. [More information](/terraform/cloud-docs/agents)
+
+
+Agent Token is a secret token that a HCP Terraform Agent is used to connect to the HCP Terraform Agent Pool.
+In `spec` only the field `Name` is allowed, the rest are used in `status`.
+More infromation:
+  - https://developer.hashicorp.com/terraform/cloud-docs/agents
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:135]: i think the previous wording was a lil bit more clear. i also updated the second sentence to be active voice and fixed typo

```suggestion
HCP Terraform agents use secret tokens, called `AgentToken`s, to connect to the HCP Terraform agent pool. In `spec`, you can only define `Name`. The other attributes are returned in `status`. 

More information:
  - https://developer.hashicorp.com/terraform/cloud-docs/agents
```

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 9]
```diff
@@ -142,74 +167,86 @@ _Appears in:_
 
 
 
-ConsumerWorkspace allows access to the state for specific workspaces within the same organization. Only one of the fields `ID` or `Name` is allowed. At least one of the fields `ID` or `Name` is mandatory. [More information](/terraform/cloud-docs/workspaces/state#remote-state-access-controls)
+ConsumerWorkspace allows access to the state for specific workspaces within the same organization.
+Only one of the fields `ID` or `Name` is allowed.
+At least one of the fields `ID` or `Name` is mandatory.
+More information:
+  - https://developer.hashicorp.com/terraform/cloud-docs/workspaces/state#remote-state-access-controls
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:174]: is there a reason why we put these on three different lines? it renders the same way

image

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 10]
```diff
@@ -142,74 +167,86 @@ _Appears in:_
...
 
 
 #### CustomPermissions
 
 
 
-Custom permissions let you assign specific, finer-grained permissions to a team than the broader fixed permission sets provide. [More information](/terraform/cloud-docs/users-teams-organizations/permissions#custom-workspace-permissions)
+Custom permissions let you assign specific, finer-grained permissions to a team than the broader fixed permission sets provide.
+More information:
+  - https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/permissions#custom-workspace-permissions
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:191]:
```suggestion
Custom permissions let you assign specific, finer-grained permissions to a team than the broader fixed permission sets provide.

More information:
  - https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/permissions#custom-workspace-permissions
```

Comment (im2nguyen) [reply website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:191]: I'm not sure I like these full URLs, why are we switching to this?

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 11]
```diff
@@ -142,74 +167,86 @@ _Appears in:_
...
 
 #### Module
 
 
 
-Module is the Schema for the modules API Module implements the API-driven Run Workflow [More information](/terraform/cloud-docs/run/api)
+Module is the Schema for the modules API
+Module implements the API-driven Run Workflow
+More information:
+  - https://developer.hashicorp.com/terraform/cloud-docs/run/api
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:240]:
```suggestion
Module is the Schema for the modules API. Module implements the API-driven run workflow.

More information:
  - https://developer.hashicorp.com/terraform/cloud-docs/run/api
```

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 12]
```diff
@@ -255,15 +292,15 @@ _Appears in:_
...
+| `name` _string_ | Name of the module that will be uploaded and executed.<br />Default: `this`. |
 | `variables` _[ModuleVariable](#modulevariable) array_ | Variables to pass to the module, they must exist in the Workspace. |
 | `outputs` _[ModuleOutput](#moduleoutput) array_ | Module outputs to store in ConfigMap(non-sensitive) or Secret(sensitive). |
-| `destroyOnDeletion` _boolean_ | Specify whether or not to execute a Destroy run when the object is deleted from the Kubernetes. Default: `false`. |
-| `restartedAt` _string_ | Allows executing a new Run without changing any Workspace or Module attributes. Example: `kubectl patch <KIND> <NAME> --type=merge --patch '{"spec": {"restartedAt": "'\`date -u -Iseconds\`'"}}'` |
+| `destroyOnDeletion` _boolean_ | Specify whether or not to execute a Destroy run when the object is deleted from the Kubernetes.<br />Default: `false`. |
+| `restartedAt` _string_ | Allows executing a new Run without changing any Workspace or Module attributes.<br />Example: kubectl patch KIND NAME --type=merge --patch '{"spec": {"restartedAt": "'\`date -u -Iseconds\`'"}}' |
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:303]:
```suggestion
| `restartedAt` _string_ | Allows executing a new Run without changing any Workspace or Module attributes.<br />Example: `kubectl patch KIND NAME --type=merge --patch '{"spec": {"restartedAt": "'\`date -u -Iseconds\`'"}}'` |
```

currently:
image

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx [hunk 13]
```diff
@@ -343,6 +386,21 @@ _Appears in:_
 | `runID` _string_ | Run ID of the latest run that updated the outputs. |
 
 
+#### PlanStatus
+
+
+
+
+
```
Comment (im2nguyen) [inline website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:394]:
```suggestion

```

**Review by im2nguyen — APPROVED** [review]
Overall looks good. Is the API reference page auto generated? There's a couple changes that I don't understand why we did -- things like "More information" and the full URL rather than just a link out to it. Seeing the full url is a slightly jarring experience. In the example below, showing them "Configuration versions" and "Run API reference" would be a much nicer experience.

Also, don't know why we're adding `<br/>` in the descriptions, it _should_ be smart enough to wrap the descriptions appropriately.

image

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx (hunk 1 above)
Comment (BrianMMcClain) [reply website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx]: Cleaned up the line breaks

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx (hunk 10 above)
Comment (BrianMMcClain) [reply website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx:191]: I'll do a quick pass to clean these link up

File: website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx (hunk 9 above)
Comment (BrianMMcClain) [reply website/docs/cloud-docs/integrations/kubernetes/api-reference.mdx]: Updated this across the doc

**Review by robin-norwood — APPROVED** [review]
lgtm!

**Pull request merged**